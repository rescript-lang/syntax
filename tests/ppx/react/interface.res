module A = {
  @react.component
  let make = (~x) => React.string(x)
}

module NoProps = {
  @react.component
  let make = () => <div />
}