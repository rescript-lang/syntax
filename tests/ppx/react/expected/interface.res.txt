module A = {
  type props<'x> = {x: 'x}
  let make = ({x, _}: props<_>) => React.string(x)
  let make = {
    let \"Interface$A" = (props: props<_>) => make(props)
    \"Interface$A"
  }
}

module NoProps = {
  type props = {}

  let make = (_: props) => ReactDOM.jsx("div", {})
  let make = {
    let \"Interface$NoProps" = props => make(props)

    \"Interface$NoProps"
  }
}
